### PR TITLE
Issue 73792: Removed mentions of R() and M()

### PR DESCRIPTION
**Pull request by lmalivert**
##### SUMMARY
Removed mentions of R() and M() formats.

Fixes issue #73792.

##### ISSUE TYPE
- Docs Pull Request

##### COMPONENT NAME
docs/docsite/rst/dev_guide/developing_modules_documenting.rst

### CODE REVIEW

**Review by samccann — APPROVED** [review]
LGTM

File: docs/docsite/rst/dev_guide/developing_modules_documenting.rst [hunk 1]
```diff
@@ -249,16 +247,13 @@ content in a uniform way:
 
 .. note::
 
-  For links between modules and documentation within a collection, you can use any of the options above. For links outside of your collection, use ``R()`` if available. Otherwise, use ``U()`` or ``L()`` with full URLs (not relative links). For modules, use ``M()`` with the FQCN or ``ansible.builtin`` as shown in the example. If you are creating your own documentation site, you will need to use the `intersphinx extension <https://www.sphinx-doc.org/en/master/usage/extensions/intersphinx.html>`_ to convert ``R()`` and ``M()`` to the correct links.
-
+  For links between modules and documentation within a collection, you can use any of the options above. Use ``U()`` or ``L()`` with full URLs (not relative links).
```
Comment (acozine) [inline docs/docsite/rst/dev_guide/developing_modules_documenting.rst]:
```suggestion
  For links between modules and documentation within a collection, you can use either of the options above. Use ``U()`` or ``L()`` with full URLs (not relative links).
```

**Review by acozine — COMMENTED** [review]
Looks great, thanks @lmalivert. One small nit, then it's ready to merge.

**Pull request merged**